cli/src/filter.rs

```diff
@@ -2,7 +2,50 @@ use std::collections::HashSet;
 use std::iter::FromIterator;
 use std::str::FromStr;
 
-use phylum_types::types::package::{RiskLevel, RiskType};
+use phylum_types::types::job::JobStatusResponse;
+use phylum_types::types::package::{Package, PackageStatusExtended, RiskLevel, RiskType};
+
+/// Remove issues based on a filter.
+pub trait FilterIssues {
+    fn filter(&mut self, filter: &Filter);
+}
+
+impl FilterIssues for Package {
+    fn filter(&mut self, filter: &Filter) {
+        self.issues.retain(|issue| !should_filter_issue(filter, issue.impact, issue.risk_type));
+    }
+}
+
+impl<T: FilterIssues> FilterIssues for JobStatusResponse<T> {
+    fn filter(&mut self, filter: &Filter) {
+        for package in &mut self.packages {
+            package.filter(filter);
+        }
+    }
+}
+
+impl FilterIssues for PackageStatusExtended {
+    fn filter(&mut self, filter: &Filter) {
+        self.issues.retain(|issue| !should_filter_issue(filter, issue.severity, issue.domain));
+    }
+}
+
+/// Check if a package should be filtered out.
+fn should_filter_issue(filter: &Filter, level: RiskLevel, risk_type: impl Into<RiskType>) -> bool {
```

Comment:
How is it unusual? The feature exists specifically for this purpose.

Reply:
Fair enough. [The book](https://doc.rust-lang.org/book/ch10-02-traits.html#trait-bound-syntax) agrees with you.

> The `impl Trait` syntax is convenient and makes for more concise code in simple cases, while the fuller trait bound syntax can express more complexity in other cases.

I guess I'm just not used to it for some reason... Ignore this comment.

Reply:
I mean it was introduced fairly recently, just a couple years back. But the goal is specifically to avoid having to add type parameters every time when it is not necessary.

There's still good reasons to prefer generics in many cases, `impl Trait` can't be used to specify the exact same type twice for example (`(x: T, y: T)`) and the type can't be manually specified by the caller (`foo::<Xyz>(x)`). But since none of them are relevant here, there's no reason not to use the shorthand.